### DISCUSSION TITLE
Attach Error Boundary Events to serviceTask/scriptTask

### DISCUSSION THREAD

**Discussion by Irene-UXXI** [flowsforapex/apex-flowsforapex#718] [General]
Hi All,
I am working in flows for APEX.

I would like to attach a error boundary Events to Service task so that the task's exceptions would be caught with the error.
How can I throw those errors from my task? 
I have checked with RAISE but the task appears with an error.

Thanks a lot.

**Comment by rallen2010** [comment]
HI Irene,   Just to check, if I understand correctly,   you want to be able to raise a named exception inside your PL/SQL code which then causes a BPMN error or escalation event to fire, which you can manage in the process diagram from a scriptTask or serviceTask or businessRulesTask (any task that can run user-defined PL/SQL code)?

It's on our list - will try to get it into the upcoming 24.1 release if we can...

There's probably a (messy) workaround that you could use (temporarily) today, involving creating a messageCatch event, & using your exception inside your code to throw a message to the catcher.   It's not ideal, but if you really need a workaround before the next release, you could explore it...

**Comment by rallen2010** [comment]
See Issue #657

**Comment by Irene-UXXI** [comment]
Thank you very much for the quick answer.

Yes, that's exactly the problem.
We will wait for the solution in version 24.1 and meanwhile we will apply the workaround.

**Comment by Irene-UXXI** [comment]
Hi @rallen2010 ,
Have you fixed this issue into version 24.1?

**Reply by rallen2010** [reply]
Apologies, but no it didn't make it.  We considered squeezing it in at the last minute but discovered it wasn't quite as simple as I hoped and therefore too risky at the end of the cycle when we're a bit overdue anyhow!

**Reply by Irene-UXXI** [reply]
Thank you very much for your response. No problem with waiting; for now, we're applying the workaround.

**Comment by rallen2010** [comment]
Looking at this deeper, I think this only makes sense for error events, which are by definition always interrupting.  Raising an exception in your script, so that it returns to the Flows engine will always be interrupting.   So I'm changing this to just error events, which will be on all PL/SQL script tasks (ScriptTask, BusinessRuleTask, ServiceTask, SendTask).

**Comment by rallen2010** [comment]
This is implemented in Flows for APEX 25.1 (releasing soon).
